src/olympia/versions/compare.py

```diff
@@ -99,22 +108,22 @@ def __gt__(self, other):
                 return cmp == 1
         return False
 
-    def __bool__(self):
-        # Make boolean comparison use the string - to support returning False
-        # when the string is empty.
-        return bool(self.string)
+    def __ge__(self, other):
+        return self.__gt__(other) or self.__eq__(other)
+
+    def __lt__(self, other):
+        return not self.__ge__(other)
 
-    def __str__(self):
-        return self.string
+    def __le__(self, other):
+        return not self.__gt__(other)
 
 
 def version_int(version):
     """This is used for converting an app version's version string into a
     single number for comparison.  To maintain compatibility the minor parts
     are limited to 99 making it unsuitable for comparing addon version strings.
     """
-    vstr = VersionString(version)
-    vdict = dict(vstr.full_dict)
+    vdict = dict(VersionString(version).parts)
```

Comment:
I might be missing something here, but isn't this patch about making `version` always be a `VersionString`? Why do we need `VersionString(version)`?

Reply:
The changes are related to add-on versions, but `version_int` is used by _application_ version comparison code - that doesn't use `VersionString` (yet).